### ISSUE TITLE
SA1501 applied differently when used in .editorconfig vs .ruleset

### ISSUE THREAD

**Issue by kornelijepetak** [DotNetAnalyzers/StyleCopAnalyzers#3290]
There's this piece of code

```cs
if(variable == value) return;
```

**Before.**
I used to use the `.ruleset` file in the past to set up the analyzers.

```cs
// project.csproj

<PropertyGroup>
  <CodeAnalysisRuleSet>$(SolutionDir)styles.ruleset</CodeAnalysisRuleSet>
</PropertyGroup>

// styles.ruleset
<Rule Id="SA1501" Action="Error" />
```

That setup would result in this:
![image](https://user-images.githubusercontent.com/7033216/104688101-bef0ca00-5700-11eb-8342-a099a0bef883.png)

**After.**
I have since migrated to the `.editorconfig`, removed the `.ruleset` entry from the `csproj` file and I now have this:

```cs
// .editorconfig
dotnet_diagnostic.SA1501.severity = error
```

Now, this situation is no longer detected.
![image](https://user-images.githubusercontent.com/7033216/104688703-81d90780-5701-11eb-97d3-ecba3b9bfe18.png)

While I understand that SA1501 may not even deal with this particular case, as the `cause` for the `SA1501` as per docs requires braces to be present, I don't understand why it works differently with the `.ruleset` file and not with the `.editorconfig`.

I am mostly interested in a rule that would (regardless of the braces) **disallow** putting `return/continue/break` on the same line as any other flow control statement (if, while, etc.). Is there any such rule that can be used with the `.editorconfig`?

💻 **Technical context**
* Visual Studio 2019
* no change to analyzer's version between the two setups

**Comment by sharwell** [comment]
Well that's weird 😄

**Comment by ThomasMader** [comment]
I am experiencing the same problem. Moving the settings from a ruleset to an .editorconfig doesn't seem to work at all.
Maybe this is related to https://github.com/dotnet/roslyn/issues/43051#issuecomment-608442423.
If this is the case maybe it works with Visual Studio 2019 16.9.

**Comment by manfred-brands** [comment]
Rule SA1501 depends on the configuration for SA1503 [source](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501StatementMustNotBeOnASingleLine.cs#L85)

```
[*.cs]
# Statement should not be on a single line
dotnet_diagnostic.SA1501.severity = error
# Braces should not be omitted
dotnet_diagnostic.SA1503.severity = none
```

But even with this file, I can only get SA1501 to fire on `lock (args) { return args.Length; }` but not on `lock (args) return args.Length;` Indicating that somehow it doesn't detect SA1503 is suppressed.

**Comment by sharwell** [comment]
Due to the way configuration is checked, these rules may need to be configured via **.globalconfig** (from my earlier comment) and not via **.editorconfig** in order for the relation between SA1501 and SA1503 to be detected during analysis. While not intentional, do note that **.globalconfig** (and not **.editorconfig**) is the true replacement for **.ruleset** files.

**Comment by manfred-brands** [comment]
I can confirm that renaming the file to *.globalconfig* and removing the section header, the SA1501 fires in both cases.
@sharwell Is this specific to StyleCop as I haven't noticed this in the past for other analyzers. Although I don't know if any other rules are linked in this way either.

**Comment by sharwell** [comment]
Yes, this implementation detail is specific to the way StyleCop Analyzers tries to link/coordinate the behavior of SA1501 and SA1503. In hindsight this wasn't a great idea but removing the link between the two is difficult now.
